src/VirtualClient/VirtualClient.Actions/ApacheBench/ApacheBenchExecutor.cs

```diff
@@ -0,0 +1,238 @@
+
+        /// <summary>
+        /// The ApacheBench workload executor
+        /// </summary>
+        /// <param name="dependencies">Provides all of the required dependencies to the Virtual Client component.</param>
+        /// <param name="parameters">
+        /// Parameters defined in the execution profile or supplied to the Virtual Client on the command line.
+        /// </param>
+        public ApacheBenchExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters = null)
+            : base(dependencies, parameters)
+        {
+            this.systemManagement = dependencies.GetService<ISystemManagement>();
+            this.fileSystem = dependencies.GetService<IFileSystem>();
+            this.packageManager = dependencies.GetService<IPackageManager>();
+            this.processManager = this.systemManagement.ProcessManager;
+            this.SystemManagement = dependencies.GetService<ISystemManagement>();
+            this.stateManager = this.systemManagement.StateManager;
+        }
+
+        /// <summary>
+        /// Parameter defines the command line arguments to pass to the workload executable.
+        /// </summary>
+        public string CommandLine
+        {
+            get
+            {
+                string commandLine = this.Parameters.GetValue<string>(nameof(this.CommandLine));
+                if (string.IsNullOrWhiteSpace(commandLine))
+                {
+                    commandLine = "-k -n 50000 -c 10 http://localhost:80/";
+                }
+
+                return commandLine;
+            }
+        }
+
+        /// <summary>
+        /// Provides methods for managing system requirements.
+        /// </summary>
+        protected ISystemManagement SystemManagement { get; }
+
+        /// <summary>
+        /// It is common to use local member variables or properties to keep track of the names of 
+        /// workload binaries/executables. Depending upon the OS platform (Linux vs. Windows) we are on
+        /// the names of the binaries might be different.
+        /// </summary>
+        protected string WorkloadExecutablePath { get; set; }
+
+        /// <summary>
+        /// Executes the workload.
+        /// </summary>
+        /// <param name="telemetryContext">Provides context information that will be captured with telemetry events.</param>
+        /// <param name="cancellationToken">A token that can be used to cancel the operation.</param>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            try
+            {
+                Task.Delay(3000);
+                DateTime startTime = DateTime.UtcNow;
+                string workloadResults = string.Empty;
+                using (IProcessProxy process = await this.ExecuteCommandAsync(this.WorkloadExecutablePath, this.CommandLine, string.Empty, telemetryContext, cancellationToken, runElevated: true))
```

Comment:
could u plz have a look at sample here and log process details and throw if workload fails